x-pack/plugins/spaces/public/ui_api/index.ts

```diff
@@ -6,23 +6,27 @@
  */
 
 import type { StartServicesAccessor } from 'src/core/public';
-import type { SpacesApiUi } from 'src/plugins/spaces_oss/public';
 
 import type { PluginsStart } from '../plugin';
 import { createRedirectLegacyUrl } from '../share_saved_objects_to_space';
+import { useSpaces } from '../spaces_context';
 import type { SpacesManager } from '../spaces_manager';
 import { getComponents } from './components';
+import type { LazyComponentFn, SpacesApiUi, SpacesApiUiComponent } from './types';
 
 interface GetUiApiOptions {
   spacesManager: SpacesManager;
   getStartServices: StartServicesAccessor<PluginsStart>;
 }
 
+export type { LazyComponentFn, SpacesApiUi, SpacesApiUiComponent };
+
 export const getUiApi = ({ spacesManager, getStartServices }: GetUiApiOptions): SpacesApiUi => {
   const components = getComponents({ spacesManager, getStartServices });
 
   return {
     components,
     redirectLegacyUrl: createRedirectLegacyUrl(getStartServices),
+    useSpaces,
```

Comment:
Thanks for this addition.

However, the `useSpaces` hook will only works for functional components (not class components). Class components will need to either have a reference of the context class to use the `contextType` property, or have access to the `context.Consumer` component.

Any objection adding a `getSpacesContextConsumer` API to `SpacesApiUiComponent`? If that's fine, I could do it directly in https://github.com/elastic/kibana/pull/109196

Reply:
Sure that's fine, thanks for adding it!